compiler/back_end/cpp/header_generator.py

```diff
@@ -1430,6 +1439,23 @@ def _offset_source_location_column(source_location, offset):
   return new_location
 
 
+def _verify_namespace_attribute(attr, source_file_name, errors):
+  if attr.name.text != attributes.Attribute.NAMESPACE:
+    return
+  namespace_value = attr.value.string_constant
+  if not re.match(_NS_RE, namespace_value.text):
+    errors.append([error.error(
+        source_file_name, namespace_value.source_location,
+        "Bad namespace, does not match expected format.")])
```

Comment:
"Does not match expected format" is pretty vague; maybe some variation of `must be a valid C++ namespace, like "abc", "abc::def", or "::abc::def::ghi" (ISO/IEC 14882:2017 enclosing-namespace-specifier)`?

Reply:
Done